features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Listing/Listing.js

```diff
@@ -66,13 +68,69 @@ class Listing extends React.Component {
-                        this.state.apis.list.map((api, i) => {
-                            return <ApiThumb key={api.id} listType={this.state.listType} api={api}/>
-                        }) : <Loading/>
-                    }
+                <div className="ibox">
+                    <div className="ibox-title">
+                        <h2 className="pull-left">
+                            All APIs
+                            <h5>All APIs visible to this account</h5>
+                        </h2>
+
+                        <Link className="pull-right btn btn-primary" to="/api/create">Create new API</Link>
+                    </div>
+                    <div className="clearfix"></div>
+                    <nav className="navbar navbar-default" role="navigation">
+                        {/* Collect the nav links, forms, and other content for toggling */}
+                        <div className="collapse navbar-collapse">
+                            <ul className="nav navbar-nav">
+                                <li>
+                                    <button type="button" className="btn">
+                                        <i className="fw fw-delete"></i>
+                                    </button>
+                                </li>
+                                <li>
+                                    <button type="button" className="btn">
+                                        <i className="fw fw-refresh"></i>
+                                    </button>
+                                </li>
+                            </ul>
+                            <div className="col-sm-6">
+                                <form className="navbar-form" role="search">
+                                    <div className="input-group">
+                                        <input type="text" placeholder="Search" className="input-sm form-control" />
+                                        <div className="input-group-btn">
+                                            <button className="btn btn-default" type="submit"><i className="glyphicon glyphicon-search" /></button>
+                                        </div>
+                                    </div>
+                                </form>
+                            </div>
+                            <ul className="nav navbar-nav navbar-right">
+                                <li><button type="button"
+                                            className={this.isActive('grid')}
+                                            onClick={() => this.setListType('grid')} > <i className="fw fw-grid"></i>
+                                </button></li>
+                                <li>
+                                    <button type="button"
+                                            className={this.isActive('list')}
+                                            onClick={() => this.setListType('list')} > <i className="fw fw-list"></i>
```

Comment:
IMHO  it is good to bind `this` to the method in the constructor rather than wrapping it with an arrow function.As in [documentation](https://facebook.github.io/react/docs/handling-events.html), there would be a performance issue otherwise.